core/src/impl/Kokkos_Core.cpp

```diff
@@ -932,6 +957,7 @@ void print_configuration(std::ostream& out, const bool detail) {
 bool is_initialized() noexcept { return g_is_initialized; }
 
 bool show_warnings() noexcept { return g_show_warnings; }
+bool tune_internals() noexcept { return g_tune_internals; }
```

Comment:
Did you mean to add in `Kokkos::`?

Reply:
I discussed this yesterday. I want `show_warnings` and `tune_internals` to work the same way, they both query something about Kokkos. I'm comfortable with either of two options

1) Leave both in Kokkos
2) Put tune_internals in Kokkos::Impl, same for show_warnings (deprecating the show_warnings in namespace Kokkos)

Reply:
I would prefer `Impl::` and deprecate/move `show_warnings` in a separate PR.

Reply:
I think it makes some sense to have them in Kokkos:: i.e. to let users check whether this is on or not.

Reply:
I have no strong feelings one way or the other. You two have a discussion and see how you feel?

Reply:
I thought a bit more about it, and I'd like to leave the function in namespace Kokkos (no impl). Some things a user might do

"If the user didn't tell Kokkos to tune, I won't tune either" 
"if the user told Kokkos to tune, I don't need to tune such-and-such a parameter because it's analogous to one in Kokkos"